urllib3/_collections.py

```diff
@@ -24,7 +59,7 @@ def __exit__(self, exc_type, exc_value, traceback):
 _Null = object()
 
 
-class RecentlyUsedContainer(MutableMapping):
+class RecentlyUsedContainer(collections.MutableMapping, Generic[_K, _V]):
```

Comment:
We can also inherit from `typing.MutableMapping[_K, _V]`, but it is not working just yet, require some change in the `typing` module.